test/org/zaproxy/zap/extension/pscanrulesAlpha/FeaturePolicyScannerUnitTest.java

```diff
@@ -0,0 +1,121 @@
+/*
+ * Zed Attack Proxy (ZAP) and its related class files.
+ *
+ * ZAP is an HTTP/HTTPS proxy for assessing web application security.
+ *
+ * Copyright 2019 The ZAP development team
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.zaproxy.zap.extension.pscanrulesAlpha;
+
+import static org.junit.Assert.assertEquals;
+
+import org.apache.commons.httpclient.URI;
+import org.apache.commons.httpclient.URIException;
+import org.junit.Before;
+import org.junit.Test;
+import org.parosproxy.paros.Constant;
+import org.parosproxy.paros.network.HttpMessage;
+import org.parosproxy.paros.network.HttpRequestHeader;
+
+public class FeaturePolicyScannerUnitTest extends PassiveScannerTest<FeaturePolicyScanner> {
+	
+	private static final String MESSAGE_PREFIX = "pscanalpha.featurepolicymissing.";
+	private HttpMessage msg;
+
+    @Before
+    public void before() throws URIException {
+        HttpRequestHeader requestHeader = new HttpRequestHeader();
+        requestHeader.setURI(new URI("http://example.com", false));
+
+        msg = new HttpMessage();
+        msg.setRequestHeader(requestHeader);
+    }
+    
+    @Override
+    protected FeaturePolicyScanner createScanner() {
+    	return new FeaturePolicyScanner();
+    }
+	
+    /**	
+     * if an HTTP response with Content-Type = HTML does not contain feature policy then an alert should be raised
+     * @throws Exception
+     */
+    @Test
+    public void shouldRaiseAlertOnMissingFeaturePolicyHTML() throws Exception {
+        // Given
+        msg.setResponseHeader("HTTP/1.1 200 OK\r\n" + "Server: Apache-Coyote/1.1\r\n" + "Content-Type: HTML\r\n");
```

Comment:
Status could be set in “before”, server can be dropped.

Reply:
Resolved with 6bdf7f5dab2990bd992de52965560cd8fd744c3e